frontend/packages/ux-editor/src/components/TextResource/hooks/useTextResourceValue.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { useTextResourceValue } from './useTextResourceValue';
+import { textResourceByLanguageAndIdSelector } from '@altinn/ux-editor/selectors/textResourceSelectors';
+import { DEFAULT_LANGUAGE } from 'app-shared/constants';
+import { renderHook } from '@testing-library/react';
+import { ServicesContextProvider } from 'app-shared/contexts/ServicesContext';
+import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
+import React from 'react';
+import { queriesMock } from 'app-shared/mocks/queriesMock';
+
+// Test data
+const id = 'testId';
+const value = 'testValue';
+
+// Mocks
+jest.mock('@altinn/ux-editor/hooks', () => ({
+  useTextResourcesSelector: jest.fn((selector) => ({
+    value: selector === textResourceByLanguageAndIdSelector(DEFAULT_LANGUAGE, id) ? value : value,
+  })),
+}));
+
+const renderUseTextResourceValue = () => {
+  const client = createQueryClientMock();
+  return renderHook(() => useTextResourceValue(id), {
+    wrapper: ({ children }) => (
+      <ServicesContextProvider {...queriesMock} client={client}>
+        {children}
+      </ServicesContextProvider>
+    ),
+  });
+};
+
+describe('useTextResourceValue', () => {
+  it('should return the text resource value', () => {
+    const { result } = renderUseTextResourceValue();
+    expect(result.current).toEqual(value);
+  });
+});
```

Comment:
I think this is a better way to render the hook for the test. The `serviceContextProvider` is already implemented inside the renderHookWithProviders function, with all the context values it needs 😊 

```
const id = 'testId';
const value = 'testValue';
const language = 'nb';

// Mocks
const textResourcesMock = { [language]: [{ id, value }] };

describe('useTextResourceValue', () => {
  it('should return the text resource value', () => {
    const queryClient = createQueryClientMock();
    queryClient.setQueryData([QueryKey.TextResources, org, app], textResourcesMock);
    queryClient.setQueryData([QueryKey.TextLanguages, org, app], [language]);
    const { result } = renderHookWithProviders(() => useTextResourceValue(id), 
        {queryClient});
    expect(result.current).toEqual(value);
  });
});
```

Reply:
Fixed